### ISSUE TITLE
Indicate file upload restriction

### ISSUE THREAD

**Issue by elrido** [PrivateBin/PrivateBin#95]
It might be nice to display the maximum allowed upload size next to the attachment button or in the dropdown.

Counter points:
- That size applies to paste and attachment in compressed, encrypted, base64 encoded form together. Depending on the content it could be that you seemingly are able to upload more (because your content compresses well) or up to 30% less then that (because your content does not compress, the encryption makes it slightly larger and then the base64 adds its 20 - 25% on top of that).  
Although this could be counteracted by checking the length of the payload before sending in JS, so an appropriate message could be displayed.
- A server admin might not wish to disclose the upper limit as it might be abused (Well, if you want to still figure it out, you just upload larger and larger files of random data until you hit the limit).

**Comment by rugk** [comment]
Mhh, are files already checked by the JS before they are uploaded, so unnecessary uploads are avoded?

**Comment by elrido** [comment]
Not yet. At least an error in that case would make sense IMHO.

**Comment by rugk** [comment]
However if we do so, we need to transfer the server limit to the JS side, which defeats the purpose of the second point you mentioned.

**Comment by r4sas** [comment]
I will duplicate my post from #188:
>Or for example request, that will get that limit from server to check with size of data that we will send.
>API POST request to / with data, like: {'get':'sizelimit'}
>RESPONSE {'status':0,'sizelimit':512000} (sizelimit in bytes)

**Comment by r4sas** [comment]
Can be that done now in template on page building stage?

To prevent creating one more request, it can be done this way:
For example provide overall paste size limit via `meta` tag in page header. Аfter page loading find that tag (for example `<meta name="privatebin-sizelimit" content="2097152">) with JS and use value for UI and size check before sending (to prevent size exceeded pastes).

**Comment by elrido** [comment]
Yes, you can: First, expose that information from the configuration file as one of the variables available in the templates by adding an appropriate line in the [Controller.php (link points to similar example)](https://github.com/PrivateBin/PrivateBin/blob/c707c87cac67359b99349f40c776c8d301f2bf2a/lib/Controller.php#L382):
```php
$page->assign('SIZELIMIT', $this->_conf->getKey('sizelimit'));
```
The variable doesn't need to be in all capitals, that is just a convention I kept using from back in ZeroBins template engine.

The above being done, you can insert this variable anywhere in your HTML template:
```php
<?php echo $SIZELIMIT; ?>
```
